ext/nnstreamer/tensor_decoder/tensordec-imagesegment.c

```diff
@@ -0,0 +1,287 @@
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensordec-directvideo.c
+ * @date	19 Oct 2019
+ * @brief	NNStreamer tensor-decoder subplugin, "image segment",
+ *              which detects objects and paints their regions.
+ *
+ * @see		https://github.com/nnsuite/nnstreamer
+ * @author  Jihoon Lee <ulla4571@gmail.com>
+ *          niklasjang <niklasjang@gmail.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * option1: Decoder mode of image segmentation
+ *          Available : tflite-deeplab
+ */
+
+#include <string.h>
+#include <glib.h>
+#include <gst/video/video-format.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
+
+void init_is (void) __attribute__ ((constructor));
+void fini_is (void) __attribute__ ((destructor));
+
+#define RGBA_CHANNEL                   4
+#define TFLITE_DEEPLAB_TOTAL_LABELS    21
+
+/**
+ * @brief There can be different schemes for image segmentation
+ */
+typedef enum
+{
+  TFLITE_DEEPLAB = 0,
+  UNKNOWN,
+} image_segment_modes;
+
+/**
+ * @brief List of image-segmentation decoding schemes in string
+ */
+static const char *is_modes[] = {
+  [TFLITE_DEEPLAB] = "tflite-deeplab",
+  NULL,
+};
+
+/**
+ * @brief Data structure for image segmentation info
+ */
+typedef struct
+{
+  image_segment_modes mode; /**< The image segmentation decoding mode */
+  guint **segment_map;  /**< The image segmentated map */
+
+  guint width; /**< Input video width */
+  guint height; /**< Input video height */
+} image_segments;
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+is_init (void **pdata)
+{
+  image_segments *idata;
+  *pdata = g_new0 (image_segments, 1);
+
+  idata = *pdata;
+  idata->mode = UNKNOWN;
+  idata->width = 0;
+  idata->height = 0;
+  idata->segment_map = NULL;
+
+  return TRUE;
+}
+
+/** @brief Free the allocated segment_map */
+static void
+_free_segment_map (image_segments * idata)
+{
+  int i;
+
+  if (idata->segment_map) {
+    for (i = 0; i < idata->height; i++) {
+      g_free (idata->segment_map[i]);
+    }
+    g_free (idata->segment_map);
+  }
+
+  idata->segment_map = NULL;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static void
+is_exit (void **pdata)
+{
+  image_segments *idata = *pdata;
+
+  _free_segment_map (idata);
+
+  g_free (*pdata);
+  *pdata = NULL;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+is_setOption (void **pdata, int op_num, const char *param)
+{
+  image_segments *idata = *pdata;
+
+  if (op_num == 0) {
+    /* option1 = image segmentation decoder mode */
```

Comment:
option == 0 and option1 keep confusing me.
```
The first option indicates a mode of this decoder sub-plugin.
```

Or you use your own phrase. Anyway change it.